controllers/profile.js

```diff
@@ -0,0 +1,34 @@
+const Profile = require("../models/Profile");
+var ObjectId = require("mongoose").Types.ObjectId;
+const { StatusCodes } = require("http-status-codes");
+const { BadRequestError, UnauthenticatedError } = require("../errors");
+
+const getProfileData = async (req, res) => {
+  try {
+    const data = await Profile.findOne({ email: req.body.email });
+
+    res.status(StatusCodes.OK).json(data);
+  } catch (error) {
+    console.log(error);
+    return;
+  }
+};
+
+const updateProfile = async (req, res) => {
+  const { userName, city, country } = req.body.data;
+  req.body.data.email = req.body.email;
+
+  if (userName === "") {
+    throw new BadRequestError("Name can't be empty");
+  }
+  const profile = await Profile.findOneAndUpdate({ email: req.body.email }, req.body.data, {
+    new: true,
+    runValidators: true,
+  });
+  if (!profile) {

```

Comment:
**Suggestion:** Instead of updating fields from the request, extract the values you want from the body and create an object in the format you prefer.
```suggestion
const {
    email,
    data: { userName, city, country },
  } = req.body.data;

  if (userName === "") {
    throw new BadRequestError("Name can't be empty");
  }
  const profile = await Profile.findOneAndUpdate(
    { email },
    { email, userName, city, country },
    {
      new: true,
      runValidators: true,
    }
  );
```

Reply:
Another approach for this endpoint would be to get the identifier (email) for the user from the route/path instead of the body. Then your body could just have the other values direclty instead of using an extra property called `data`.

Reply:
**Remove data property from request:**
```
const response = await axios.patch(`http://localhost:3000/api/v1/profile, {
        userName: nameState,
        city: cityState,
        country: countryState,
        email: emailState,
      });
```

**Fix endpoint:**
```
  const { userName, city, country, email } = req.body;  

  if (userName === "") {
    throw new BadRequestError("Name can't be empty");
  }
  const profile = await Profile.findOneAndUpdate(
    { email },
    { userName, city, country, email },
    {
      new: true,
      runValidators: true,
    }
  );
```
Thank you! it was mess really